src/components/cc-pricing-page/cc-pricing-page.js

```diff
@@ -1,146 +1,189 @@
-    this.zones = null;
+    /** @type {Temporality} Sets the current selected temporality. */
+    this.selectedTemporality = DEFAULT_TEMPORALITY;
 
-    this._selectedPlans = {};
+    /** @type {string} Sets the current selected zone by referencing its ID/name. */
+    this.selectedZoneId = 'par';
+
+    /** @type {CcPricingHeader|null} */
+    this._headerElement = null;
+
+    /** @type {CcPricingProduct[]|null} */
+    this._productElements = null;
+
+    /** @type {CcPricingEstimation[]|null} */
+    this._estimationElement = null;
+  }
+
+  _getPlanId (plan) {
+    return plan.id ?? `${plan.productName}/${plan.name}`;
   }
 
   _getTotalPrice () {
-    return Object.values(this._selectedPlans)
+    return Object.values(this.selectedPlans)
       .map(({ price, quantity }) => price * 24 * 30 * quantity)
       .reduce((a, b) => a + b, 0);
   }
 
-  _getPlanId (plan) {
-    return plan.id ?? `${plan.productName}/${plan.name}`;
+  /**
+   * Query pricing components from light DOM and store in their corresponding state props.
+   */
+  _updateElementReferences () {
+    this._headerElement = this.querySelector('cc-pricing-header');
+    this._productElements = Array.from(this.querySelectorAll('cc-pricing-product, cc-pricing-product-consumption'));
+    this._estimationElement = this.querySelector('cc-pricing-estimation');
   }
 
   _onAddPlan ({ detail: plan }) {
     const planId = this._getPlanId(plan);
-    if (this._selectedPlans[planId] == null) {
-      this._selectedPlans[planId] = { ...plan, quantity: 0 };
+    if (this.selectedPlans[planId] == null) {
+      this.selectedPlans[planId] = { ...plan, quantity: 0 };
     }
-    this._selectedPlans[planId].quantity += 1;
+    this.selectedPlans[planId].quantity += 1;
     this.requestUpdate();
   }
 
+  _onChangeCurrency ({ detail: currency }) {
+    this.selectedCurrency = currency;
+  }
+
+  _onChangeTemporality ({ detail: temporality }) {
+    this.selectedTemporality = temporality;
+  }
+
   _onChangeQuantity ({ detail: plan }) {
     const planId = this._getPlanId(plan);
-    this._selectedPlans[planId].quantity = plan.quantity;
+    this.selectedPlans[planId].quantity = plan.quantity;
     this.requestUpdate();
   }
 
   _onDeletePlan ({ detail: plan }) {
     const planId = this._getPlanId(plan);
-    delete this._selectedPlans[planId];
+    delete this.selectedPlans[planId];
     this.requestUpdate();
   }
 
-  _onChangeCurrency ({ detail: currency }) {
-    dispatchCustomEvent(this, 'change-currency', currency);
+  /**
+  * When the component is connected to the DOM:
+  *
+  * - we update the list of slotted pricing components,
+  * - we set up a MutationObserver that will keep the list of slotted pricing components updated.
+  */
+  connectedCallback () {
+    super.connectedCallback();
+    this._updateElementReferences();
+    this._observer = new MutationObserver(() => this._updateElementReferences());
```

Comment:
thought: We could also have use a `slotchange` event on the `<slot>` element. I'm wondering which technique is the best.